python/lsst/sims/GalSimInterface/galSimInterpreter.py

```diff
@@ -1229,18 +1229,10 @@ def getGoodPhotImageSize(obj, keep_sb_level, pixel_scale=0.2):
     N = obj.getGoodImageSize(pixel_scale)
     #print('N = ',N)
 
-    try:
+    if isinstance(obj, galsim.RandomKnots):
         # If the galaxy is a RandomWalk, extract the underlying profile for this calculation
         # rather than using the knotty version, which will pose problems for the xValue function.
-        gal = obj.original.obj_list[0]
-        gal = galsim.Transformation(gal.original._profile,
-                gal.jac, gal.offset, gal.flux_ratio, gal.gsparams)
-        obj = galsim.Convolve(gal, *obj.original.obj_list[1:]) * obj.flux_ratio
-    except Exception:
-        # If not a RandomWalk, then `._profile` will raise an AttributeError.
-        # Catch any (non-Base) Exception though in case there are other possibilities
-        # I didn't anticipate, where we should just leave the obj alone.
-        pass
+        obj = obj._profile
```

Comment:
Did other parts of the code change so that `obj` is now the original galaxy object?  I thought at this point in the code, the object was a transformed version of the galaxy.  Not the original galaxy object (RandomKnots or otherwise).  That was the point of all that Transformation/Convolve stuff.

Reply:
Oh!  Actually, I bet it was a change we added to GalSim 2.2 that broke this.  Oops.  Now setting the flux doesn't wrap things in a Transformation object if it can just update the `flux` parameter of the underlying object.  I bet this was the only thing that was in the Transformation at this point previously.  So I think this does make sense!

Reply:
(I posted a comment here before, but I guess it didn't get saved when I resolved the conversation.  Anyway, here it is.)

> Did other parts of the code change so that obj is now the original galaxy object? I

Right, this was the outcome of #87 to handle the `FitsImage` case.

> Actually, I bet it was a change we added to GalSim 2.2 that broke this.

So it was broken independently by #87, and that's what caused the errors Antonio saw in his Theta testing, but even if we had update things correctly here in #87, the interface change may still have bitten us.